client/app/visualizations/map/map-editor.html

```diff
@@ -70,6 +72,91 @@
       <select ng-options="tile.url as tile.name for tile in $ctrl.mapTiles"
         ng-model="$ctrl.options.mapTileUrl" class="form-control"></select>
     </div>
-  </div>
 
+    <h4>Marker</h4>
+
+    <div class="checkbox">
+      <label>
+        <input type="checkbox" ng-model="$ctrl.options.customizeMarkers">
+        <i class="input-helper"></i>Override default style
+      </label>
+    </div>
+
+    <div class="row">
+      <div class="col-xs-6">
+        <div class="form-group">
+          <label>Shape</label>
+          <select ng-options="key as value for (key, value) in $ctrl.iconShapes"
+            ng-model="$ctrl.options.iconShape" class="form-control"
+            ng-disabled="!$ctrl.options.customizeMarkers"></select>
+        </div>
+      </div>
+      <div class="col-xs-6">
+        <div class="form-group">
+          <label>Icon Font
+            <span class="m-1-5" uib-popover-html="'Available <a href=&quot;https://fontawesome.com/v4.7.0/icons/&quot; target=&quot;_blank&quot;>icons.</a>'"
+              popover-trigger="'click outsideClick'">
+              <i class="fa fa-question-circle"></i>
+            </span>
+          </label>
+          <input ng-model="$ctrl.options.iconFont" type="text" class="form-control"
+            ng-disabled="!$ctrl.options.customizeMarkers"></select>
+        </div>
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="col-xs-6">
+        <div class="form-group">
+          <label class="control-label">Foreground Color</label>
+          <ui-select ng-model="$ctrl.options.foregroundColor"
+            ng-disabled="!$ctrl.options.customizeMarkers">
+            <ui-select-match>
+              <color-box color="$select.selected.value"></color-box>
+              <span ng-bind-html="$select.selected.key | capitalize"></span>
+            </ui-select-match>
+            <ui-select-choices repeat="color.value as (key, color) in $ctrl.colors">
+              <color-box color="color.value"></color-box>
+              <span ng-bind-html="color.key | capitalize | highlight: $select.search"></span>
```

Comment:
This is great (I assume it's due to the white color) but notice the text and color are a bit vertically misaligned.

Reply:
Help needed here. I played around a little, but I couldn't make the color box aligned for both selected-item and listed-item.

maybe two pixels up for selected-item and one pixel down for listed-item?
![image](https://user-images.githubusercontent.com/1698635/59557029-f4f03880-9009-11e9-8046-92aef8fa7888.png)

Reply:
I've tried my best. Please review.

Reply:
Here ya go https://github.com/deecay/redash/pull/2